storm-core/src/jvm/org/apache/storm/daemon/supervisor/SupervisorData.java

```diff
@@ -0,0 +1,339 @@
+        this.iSupervisor = iSupervisor;
+        this.active = true;
+        this.upTime = Utils.makeUptimeComputer();
+        this.stormVersion = VersionInfo.getVersion();
+        this.workerThreadPidsAtom = new ConcurrentHashMap<String, String>();
+        this.deadWorkers = new ConcurrentHashSet();
+
+        List<ACL> acls = null;
+        if (Utils.isZkAuthenticationConfiguredStormServer(conf)) {
+            acls = SupervisorUtils.supervisorZkAcls();
+        }
+
+        try {
+            this.stormClusterState = ClusterUtils.mkStormClusterState(conf, acls, new ClusterStateContext(DaemonType.SUPERVISOR));
+        } catch (Exception e) {
+            LOG.error("supervisor can't create stormClusterState");
+            throw Utils.wrapInRuntime(e);
+        }
+
+        try {
+            this.localState = ConfigUtils.supervisorState(conf);
+            this.localizer = Utils.createLocalizer(conf, ConfigUtils.supervisorLocalDir(conf));
+        } catch (IOException e) {
+            throw Utils.wrapInRuntime(e);
+        }
+        this.supervisorId = iSupervisor.getSupervisorId();
+        this.assignmentId = iSupervisor.getAssignmentId();
+
+        try {
+            this.hostName = Utils.hostname(conf);
+        } catch (UnknownHostException e) {
+            throw Utils.wrapInRuntime(e);
+        }
+
+        this.currAssignment = new ConcurrentHashMap<>();
+
+        this.heartbeatTimer = new StormTimer(null, new DefaultUncaughtExceptionHandler());
+
+        this.eventTimer = new StormTimer(null, new DefaultUncaughtExceptionHandler());
+
+        this.blobUpdateTimer = new StormTimer("blob-update-timer", new DefaultUncaughtExceptionHandler());
+
+        this.assignmentVersions = new ConcurrentHashMap<>();
+        this.syncRetry = new AtomicInteger(0);
+        this.stormIdToProfileActions = new ConcurrentHashMap<>();
+        if (Utils.getBoolean(conf.get(Config.STORM_RESOURCE_ISOLATION_PLUGIN_ENABLE), false)) {
+            try {
+                this.resourceIsolationManager = (CgroupManager) Utils.newInstance((String) conf.get(Config.STORM_RESOURCE_ISOLATION_PLUGIN));
+                this.resourceIsolationManager.prepare(conf);
+                LOG.info("Using resource isolation plugin {} {}", conf.get(Config.STORM_RESOURCE_ISOLATION_PLUGIN), resourceIsolationManager);
+            } catch (IOException e) {
+                throw Utils.wrapInRuntime(e);
+            }
+        } else {
+            this.resourceIsolationManager = null;
+        }
+    }
+
+    public ConcurrentHashMap<String, List<ProfileRequest>> getStormIdToProfileActions() {
+        return stormIdToProfileActions;
+    }
+
+    public void setStormIdToProfileActions(Map<String, List<ProfileRequest>> stormIdToProfileActions) {
+        this.stormIdToProfileActions.clear();
+        this.stormIdToProfileActions.putAll(stormIdToProfileActions);
+    }
+
+    public Map getConf() {
+        return conf;
+    }
+
+    public void setConf(Map conf) {
```

Comment:
This is not used and I am not sure of a time we would want to do this.  Can we remove setConf and make conf final?